### PR TITLE
MAYA-112175 Test rotate pivot translation.

### PR DESCRIPTION
**Pull request by williamkrick**

### CODE REVIEW

File: test/lib/ufe/testRotatePivot.py [hunk 1]
```diff
@@ -158,12 +158,14 @@ def testRotatePivotCmd(self):
...
         pivot = om.MPoint(-10, 0, 0)
+        pivotTranslate = om.MPoint(0, 0, 0)
         xyWorldValue = sin(rotZ) * 10
+        rotatedPosition = [xyWorldValue, xyWorldValue, 0]
 
         # USD sphere is at (10, 0, 0) in local space, and since its parents
         # have an identity transform, in world space as well.
         spherePath = ufe.PathString.path(
-                '|usdSphereParent|usdSphereParentShape,/sphereXform/sphere')
+                '|usdSphereParent|usdSphereParentShape,/sphereXform')
```
Comment (williamkrick) [inline test/lib/ufe/testRotatePivot.py:168]: This is a minor change but it means that all the work is done on /sphereXform, which already has some translation. Keeping all the transform stack together makes sense to me.

File: test/lib/ufe/testRotatePivot.py [hunk 2]
```diff
@@ -176,15 +178,31 @@ def testRotatePivotCmd(self):
...
+        # MAYA-112175, otherwise a zero initial rotation means rotate pivot
+        # translation will be empty and we get the correct result by accident.
+        if (mayaUtils.previewReleaseVersion() >= 127):
+            cmds.rotate(0, 0, 90)
+            pivot = om.MPoint(0, 10, 0)
+            pivotTranslate = om.MPoint(-10, -10, 0)
+            rotatedPosition = [xyWorldValue, -xyWorldValue, 0]
 
-        cmds.move(-12, 0, 0, relative=True, ufeRotatePivot=True)
+        cmds.move(-10, 0, 0, relative=True, ufeRotatePivot=True)
```
Comment (williamkrick) [inline test/lib/ufe/testRotatePivot.py:194]: Move by 10 instead of 12, already moved back to the origin.

File: test/lib/ufe/testRotatePivot.py [hunk 3]
```diff
@@ -176,15 +178,31 @@ def testRotatePivotCmd(self):
         t3d.rotatePivot(2, 0, 0)
         usdPivot = t3d.rotatePivot()
         self.assertEqual(v3dToMPoint(usdPivot), om.MPoint(2, 0, 0))
+        t3d.rotatePivot(0, 0, 0)
+
+        print(mayaUtils.previewReleaseVersion())
```
Comment (ppt-adsk) [inline test/lib/ufe/testRotatePivot.py:183]: You can remove this debugging statement.

File: test/lib/ufe/testRotatePivot.py [hunk 4]
```diff
@@ -193,7 +211,16 @@ def testRotatePivotCmd(self):
...
         sphereMatrix = om.MMatrix(t3d.inclusiveMatrix().matrix)
-        self.checkPos(sphereMatrix, [xyWorldValue, xyWorldValue, 0])
+        self.checkPos(sphereMatrix, rotatedPosition)
+
+        if (mayaUtils.previewReleaseVersion() >= 127):
+            cmds.undo()
+            cmds.move(10, 10, 0, absolute=True)
+            sphereMatrix = om.MMatrix(t3d.inclusiveMatrix().matrix)
+            self.checkPos(sphereMatrix, [10, 10, 0])
+            cmds.move(10, 10, 0, absolute=True, rotatePivotRelative=True)
```
Comment (williamkrick) [inline test/lib/ufe/testRotatePivot.py:221]: This move is the one that does not work correctly without my Maya changes.

**Review by ppt-adsk — APPROVED** [review]

**Review by ppt-adsk — APPROVED** [review]

**Review by ppt-adsk — APPROVED** [review]

**Pull request merged**